.github/workflows/ci.yml

```diff
@@ -86,10 +86,33 @@ jobs:
       - name: Install compiler tools on macOS
         if: runner.os == 'macOS'
         run: |
-          brew install make automake swig gmp mpfr boost
+          brew install make automake swig mpfr boost
           export PATH="/usr/local/opt/make/libexec/gnubin:$PATH"
 
-      - uses: pypa/cibuildwheel@v2.17
+      - name: Clone gmp
+        if: runner.os == 'macOS'
+        uses: actions/checkout@v4
+        with:
+          repository: gmp-mirror/gmp-6.3
+          path: gmp-6.3
+
+      - name: Build gmp from source for MACOSX_DEPLOYMENT_TARGET
+        if: runner.os == 'macOS'
+        env:
+          MACOSX_DEPLOYMENT_TARGET: "10.9"
+        working-directory: gmp-6.3
+        run: |
+          brew install texinfo
+          autoreconf -i -f
+          ./configure --help
+          ./configure \
+            --prefix=$(brew --prefix) \
+            --enable-cxx
+          touch ./doc/version.texi
+          make --jobs
+          make install
```

Comment:
@henryiii tagging you for review to see if you know of a more intelligent/faster way to build `gmp` for source to avoid `MACOSX_DEPLOYMENT_TARGET` mismatch.

Reply:
This is more or less what you need to do. If you use CMake, you can make it an option to have it do this for you, but otherwise, it's fine.